### PR TITLE
feat(team-workflow): Update `bulk_subscribe` to accept teams

### LINKED ISSUE #55556: Add subscription for team in `models/groupsubscription`
Two updates we need to make here:
1) Update `subscribe_actor` (https://github.com/getsentry/sentry/blob/master/src/sentry/models/groupsubscription.py#L67)  to no longer read all the team's users and just set subscribe the Team
2) Update `bulk_subscribe` to allow for bulk subscribing teams, instead of just users

### PR DESCRIPTION
**Pull request by isabellaenriquez**
This PR updates `bulk_subscribe` so that it can accept both teams and user IDs (or either).

~~❗ BLOCKED BY #55879 and #55825~~

Closes #55556

### CODE REVIEW

**Comment by isabellaenriquez** [comment]
a bunch of tests will fail rn just bc of the blocking tickets, so dw 😄😗

File: src/sentry/models/groupsubscription.py
Comment (JoshFerge) [inline src/sentry/models/groupsubscription.py:149]: you could use `teams.difference(existing_team_subscriptions)` instead of a list comprehension to filter since they're sets https://python-reference.readthedocs.io/en/latest/docs/sets/difference.html

Comment (isabellaenriquez) [reply src/sentry/models/groupsubscription.py:149]: The issue is that `existing_team_subscriptions` only returns a set of the team ids, rather than a set of teams. So the difference between `teams` and `existing_team_subscriptions` is equivalent to `teams` as nothing gets filtered out. Additionally, the same logic is used to deal with users.

File: src/sentry/models/groupsubscription.py
Comment (snigdhas) [inline src/sentry/models/groupsubscription.py:104]: Python's one-liners are great for this sort of thing 

```suggestion
        user_ids = set(user_ids) if user_ids else set()
```

File: src/sentry/models/groupsubscription.py
Comment (snigdhas) [inline src/sentry/models/groupsubscription.py:110]:
```suggestion
        # Unique the teams.
        teams = set(teams) if teams else set()
```

File: tests/sentry/models/test_groupsubscription.py
Comment (snigdhas) [inline tests/sentry/models/test_groupsubscription.py:50]: we can use `_` for variables we don't actually use

```suggestion
        for _ in range(20):
```

File: tests/sentry/models/test_groupsubscription.py
Comment (snigdhas) [inline tests/sentry/models/test_groupsubscription.py:89]: great comment!

File: src/sentry/models/groupsubscription.py
Comment (roggenkemper) [inline src/sentry/models/groupsubscription.py:127]: i think i changed my mind and we should add in a feature flag for this code inside bulk_subscribe. i don't anticipate it causing any problems, but i think we should err on the side of caution

Comment (isabellaenriquez) [reply src/sentry/models/groupsubscription.py:127]: fair enough, will make the change in a sec

**Review by roggenkemper — APPROVED** [review]
lgtm , but let's get one more approval before we merge

**Review by snigdhas — APPROVED** [review]

**Pull request merged**
